node/src/checkpointers.test.ts

```diff
@@ -0,0 +1,145 @@
+            const blockNumber = BigInt(101);
+            const checkPointerInstance = await checkpointer.getInstance();
+            await checkPointerInstance.checkpoint(blockNumber,'txn1');
+            expect(checkPointerInstance.getBlockNumber()).toEqual(blockNumber);
+            return expect(checkPointerInstance.getTransactionIds()).toStrictEqual(new Set(['txn1']));
+        });
+
+        it('Checkpointing only a new block in used checkpointer gives new block number and no transactions', async () => {
+            const blockNumber1 = BigInt(101);
+            const blockNumber2 = BigInt(102);
+            const checkPointerInstance = await checkpointer.getInstance();
+            await checkPointerInstance.checkpoint(blockNumber1, 'txn1');
+            await checkPointerInstance.checkpoint(blockNumber2);
+            expect(checkPointerInstance.getBlockNumber()).toStrictEqual(blockNumber2);
+            expect(checkPointerInstance.getTransactionIds()).toStrictEqual(new Set());
+        });
+
+        it('Checkpointing new block and transaction in used checkpointer gives new block and only new transaction', async () => {
+            const blockNumber1 = BigInt(101);
+            const blockNumber2 = BigInt(102);
+            const checkPointerInstance = await checkpointer.getInstance();
+
+            await checkPointerInstance.checkpoint(blockNumber1, 'txn1');
+            await checkPointerInstance.checkpoint(blockNumber2, 'txn2');
+            await checkPointerInstance.checkpoint(blockNumber2, 'txn3');
+            expect(checkPointerInstance.getBlockNumber()).toStrictEqual(blockNumber2);
+            expect(checkPointerInstance.getTransactionIds()).toStrictEqual(new Set([ 'txn2','txn3']));
+        });
+    });
+})
+
+
+describe('File Checkpointer: Test file creation and initialization', () => {
+
+    beforeAll(() => {
+        if (!fs.existsSync(dir)){
+            fs.mkdirSync(dir);
+        }
+    })
+
+    afterAll( ()=>{
+        if (fs.existsSync(dir)){
+            fs.rmSync(dir, { recursive: true, force: true });
+        }
+    })
+
+    it('In the absence of a checkpointer file , a new one get generated', async () => {
+        expect(fs.existsSync(checkPointerFilePath)).toEqual(false);
+        await getFileCheckpointerInstance();
+        expect(fs.existsSync(checkPointerFilePath)).toEqual(true);
+    });
+
+    it('checkpointer loads the already existing state', async () => {
+        const checkpointerstate : CheckpointerState = {
+            blockNumber : '101',
+            transactionIDs: []
+        }
+        //write state to file checkpointer
+        await writeToFile(checkpointerstate,checkPointerFilePath);
+
+        const checkPointerInstance = await getFileCheckpointerInstance();
+        expect(checkPointerInstance.getBlockNumber()).toEqual(BigInt(101));
+        expect(checkPointerInstance.getTransactionIds().size).toEqual(0);
+    });
```

Comment:
Instead of writing the state to file directly (which requires the tests to know the persistent data format), it might be better to create a file checkpointer and checkpoint a specific state (to create the file content), then create a new file checkpointer pointing to the same file